lib/codegen/src/isa/x86/abi.rs

```diff
@@ -161,7 +185,24 @@ pub fn allocatable_registers(_func: &ir::Function, flags: &shared_settings::Flag
 /// Get the set of callee-saved registers.
 fn callee_saved_gprs(flags: &shared_settings::Flags) -> &'static [RU] {
     if flags.is_64bit() {
-        &[RU::rbx, RU::r12, RU::r13, RU::r14, RU::r15]
+        if flags.call_conv() == CallConv::WindowsFastcall {
+            // "registers RBX, RBP, RDI, RSI, RSP, R12, R13, R14, R15 are considered nonvolatile
+            //  and must be saved and restored by a function that uses them."
+            // as per https://msdn.microsoft.com/en-us/library/6t169e9c.aspx
+            &[
+                RU::rbx,
+                /*RU::rbp,*/
+                RU::rdi,
+                RU::rsi,
+                /*RU::rsp,*/
```

Comment:
Yeah, I'll add a comment explaining that they are somewhat handled "by definition" (they are defacto restored after the return after the call).

@sunfishcode 
Since I for some reason am unable to respond to the comment above (Github Bug that doesn't show a reply box?),
so you'd like "common" insteadof system_v?

Reply:
Yes. We can clarify the meaning with comments.